apps/admin/frontend/src/utils/reporting.ts

```diff
@@ -103,39 +108,79 @@ export function generateTitleForReport({
     if (scannerId) {
       return ok(`Scanner ${scannerId} ${reportType} Report`);
     }
-  }
 
-  if (reportRank === 2) {
-    if (precinctId && votingMethod) {
+    if (partyId) {
```

Comment:
A lot of the churn here is just moving things around to make this a little more readable, may wan to review side by side.